homeassistant/components/light/mqtt.py

```diff
@@ -87,6 +105,14 @@
     vol.Optional(CONF_RGB_STATE_TOPIC): mqtt.valid_subscribe_topic,
     vol.Optional(CONF_RGB_VALUE_TEMPLATE): cv.template,
     vol.Optional(CONF_STATE_VALUE_TEMPLATE): cv.template,
+    vol.Optional(CONF_HUE_COMMAND_TOPIC): mqtt.valid_publish_topic,
+    vol.Optional(CONF_HUE_STATE_TOPIC): mqtt.valid_subscribe_topic,
+    vol.Optional(CONF_HUE_VALUE_TEMPLATE): cv.template,
+    vol.Optional(CONF_HUE_SCALE, default=DEFAULT_HUE_SCALE):
+        vol.All(vol.Coerce(int), vol.Range(min=1)),
+    vol.Optional(CONF_SATURATION_COMMAND_TOPIC): mqtt.valid_publish_topic,
+    vol.Optional(CONF_SATURATION_STATE_TOPIC): mqtt.valid_subscribe_topic,
```

Comment:
I don't think hue and saturation should be two separate topics as XY and RGB both also have all components in a single topic/template.

Also, hue without saturation doesn't make much sense. And saturation without hue doesn't do so either.

Reply:
SmartThings Mqtt detects and transmits changes in a single theme of hue and saturation. And so is the command.